### PR TITLE
Ui decoupling maybe donno

### PR DESCRIPTION
**Pull request by Taumic7**
Refactor UI.
UI with ECS test are now in game test project.
Changed some function to not use anything from ECS

### CODE REVIEW

File: Game/GameMain/UI/InitUI.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	ecs::systems::UIBitmapSystem* UIBitmapSys;
+	ecs::systems::UITextSystem* UITextSys;
+	ecs::systems::UIDebugSystem* UIDebugSys;
+	ecs::systems::UIPostRenderSystem* UIpostSys;
+};
+
+void InitUI(ecs::EntityComponentSystem& rECS, Direct2D** D2D, TempUISystemPtrs& rSystemPointers)
+{
+	rSystemPointers.UIpreSys	= rECS.createSystem<ecs::systems::UIPreRenderSystem>(0);
+	rSystemPointers.UIBitmapSys = rECS.createSystem<ecs::systems::UIBitmapSystem>();
```
Comment (Wessebanan) [inline Game/GameMain/UI/InitUI.h]: booooooooooo

File: UI/UI_test/test.cpp [hunk 2]
```diff
@@ -173,32 +96,17 @@ TEST(Expectedfalse, FailLoadImageToBitmap)
...
-	ID myID;
+	//ID myID;
 	HRESULT hr;
 	RECT rectTest = { 0,0,800,600 };
 	char hehe[10] = "pepe";
 	Dtest.CreateHwndRenderTarget(windowHandle, &rectTest);
 	hr = Dtest.LoadImageToBitmap("fdjak", hehe);
-	myID = Dtest.GetBitmapIDFromName(hehe);
-	void* jej = Dtest.GetBitmap(myID);
+	void* jej = Dtest.GetBitmap(hehe);
```
Comment (Wessebanan) [inline UI/UI_test/test.cpp]: camelCase --> snake_case 😠

File: UI/UI_test/test.cpp [hunk 3]
```diff
@@ -56,98 +54,23 @@ TEST(ExpectedTrue, LoadImageToBitmap)
...
 	Dtest.CreateHwndRenderTarget(windowHandle, &rectTest);
 	hr = Dtest.LoadImageToBitmap("PepeLaugh.jfif", hehe);
-	myID = Dtest.GetBitmapIDFromName(hehe);
-	void* jej = Dtest.GetBitmap(myID); //Using the ID to get the bitmap
+	void* jej = Dtest.GetBitmap(hehe); //Using the name to get the bitmap
 
 	EXPECT_EQ(hr, S_OK);
 	EXPECT_NE(jej, nullptr);
 
 }
```
Comment (Wessebanan) [inline UI/UI_test/test.cpp]: camelCase --> snake_case 👿

File: Game/GameMain/UI/InitUI.h [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+	rSystemPointers.UIBitmapSys = rECS.createSystem<ecs::systems::UIBitmapSystem>(9);
+	rSystemPointers.UITextSys	= rECS.createSystem<ecs::systems::UITextSystem>(9);
+	rSystemPointers.UIrectSys	= rECS.createSystem<ecs::systems::UIRectSystem>(9);
+	rSystemPointers.UIDebugSys	= rECS.createSystem<ecs::systems::UIDebugSystem>(9);
+	rSystemPointers.UIpostSys	= rECS.createSystem<ecs::systems::UIPostRenderSystem>(9);
+	my_d2d = new Direct2D;
+
+	ID3D11Texture2D* backbuffer;
+	IDXGISwapChain1* swapchain;
+	IDXGIDevice* dxgi_device;
```
Comment (NistBox) [inline Game/GameMain/UI/InitUI.h]: Naming conventions, pointer start with 'p'

File: Game/GameMain/UI/UISystems.h [hunk 5]
```diff
@@ -108,8 +107,8 @@ namespace ecs
 
 			void act(float _delta) override
 			{
-				mpD2D->getHwndRenderTarget()->BeginDraw();
-				mpD2D->getHwndRenderTarget()->Clear();
+				mpD2D->GetpContext()->BeginDraw();
+				//mpD2D->GetpContext()->Clear();
```
Comment (NistBox) [inline Game/GameMain/UI/UISystems.h]: Remove?

File: Game/GameMain/gameAI/AISystems.h [hunk 6]
```diff
@@ -779,6 +781,33 @@ namespace ecs
 					p_army = static_cast<ecs::components::ArmyComponent*>(it.next());
 					i++;
 				}
+
+
+				// Chagne the command on the UI
+				ecs::BaseComponent* p_base_comp = getComponentFromKnownEntity<ecs::components::UITextComponent>(p_army->getEntityID());
+
+				ecs::components::UITextComponent* text_comp = static_cast<ecs::components::UITextComponent*>(p_base_comp);
```
Comment (NistBox) [inline Game/GameMain/gameAI/AISystems.h]: Naming convention, pointer start with 'p'

File: Game/GameMain/gameAI/AISystems.h [hunk 7]
```diff
@@ -779,6 +781,33 @@ namespace ecs
...
+					text_comp->mStrText = "MOVE";
+					break;
+				default:
+					text_comp->mStrText = "no case for state";
+					break;
+				}
+
+
 				//Loop through the players units and remove their old state component.
 				ecs::Entity* unit;
```
Comment (NistBox) [inline Game/GameMain/gameAI/AISystems.h]: Not new code, but naming convention: pointer start with 'p'.

**Pull request merged**